### ISSUE TITLE
default.wallet location issue

### ISSUE THREAD

**Issue by etranger05** [AntelopeIO/DUNE#9]
## Overview
Found a wallet issue when creating an account.
This issue had also reported at the previous mandel DUNE repository but it looks not resolved yet.
https://github.com/eosnetworkfoundation/DUNE/issues/15

## Environment
macOS 12.5 (M1 Apple Silicon)
DUNE v1.0.0.rc2
Docker Desktop 4.11.1 (84025)
Docker Engine 20.10.17
Python 3.9.13
SHELL: zsh

## Reproduce steps
Install Docker & Python3, clone the DUNE repo, set the PATH, run bootstrap.sh as described in README.md.
And created a node and then tried to create an account.
```
$ dune --start test_node
Creating docker container [dune_container]
Creating node [test_node]
Using Configuration [/home/www-data/config.ini]

$ dune --create-account nodeone
Creating account [nodeone] with key pair [Private: 5JhPmsEtXucrmV4egq14zTqK478eRniNG2pwFmAGTNAi9BNUiQ7, Public: EOS5bNRt36PQkU7cXsbkH9PiFo6a5wYbz5VrYexQR1zTm1KibCVHP]
Error 3120006: No available wallet
Ensure that you have created a wallet and have it open
Error Details:
You don't have any wallet!
Stack Trace:
wallet_manager.cpp:140 get_public_keys

$ dune --version
DUNE v1.0.0.rc2
```
In my analysis, this is caused by the script setup_system.sh which creates wallet run as root permission by the Dockerfile, and eosio-wallet is generated under /root.

But the dune script is running as the user www-data so that it is trying to find default.wallet from /home/www-data.

This is the wallet location just after run the bootstrap.sh and create the container.
```
$ docker exec -u root -it dune_container ls -alh /root/eosio-wallet
total 16K
drwxr-xr-x 2 root root 4.0K Jul 15 12:48 .
drwx------ 1 root root 4.0K Jul 15 12:48 ..
-rw-r--r-- 1 root root 2.9K Jul 15 12:48 config.ini
-rw------- 1 root root  311 Jul 15 12:48 default.wallet
-rw-r--r-- 1 root root    0 Jul 15 12:48 wallet.lock

$ docker exec -u root -it dune_container ls -alh /home/www-data
total 16K
drwxr-xr-x 1 www-data www-data 4.0K Jul 15 12:49 .
drwxr-xr-x 1 root     root     4.0K Jul 15 12:49 ..
-rw-r--r-- 1 www-data www-data  621 Jul 15 12:49 config.ini
drwxr-xr-x 2 www-data www-data 4.0K Jul 15 12:49 nodes
```
I tried to make a PR but it looks a bit complicated as it may related to permission issue and other features such as export/import wallet.

## Workaround
so I'm just leaving commands for somebody who need a simple workaround as below. You can run it just after bootstrap.sh completed.
```
$ dune --create-account <username> # this will fail but generate eosio-wallet structure under /home/www-data.
$ docker exec -u root -it dune_container cp /root/eosio-wallet/default.wallet /home/www-data/eosio-wallet/ 
$ docker exec -u root -it dune_container chown www-data:www-data /home/www-data/eosio-wallet/default.wallet
$ dune --create-account <username> # this will create the account.
```

Thanks.

**Comment by jolly-fellow** [comment]
I'd recommend to check this bug again. I believe it may be fixed in #34. I just checked it on Ubuntu it works well. Unfortunately I have no Mac to test there. By the way Docker Desktop 4.13.0 and 4.13.1 have several bug fixes related to the Docker file sharing system in MacOS.

**Comment by jolly-fellow** [comment]
During work on #53 I have reviewed the code of implementations of keys --start  and --create-account, corrected the paths and tested these keys on Ubuntu. Everything works well so I'm closing this issue.
